apps/web/src/utils/getHookByAddress.ts

```diff
@@ -11,3 +13,14 @@ export const getHooksMap = memoize((chainId: number) => {
 export const getHookByAddress = (chainId?: ChainId, address?: HookData['address']): HookData | undefined => {
   return chainId && address ? getHooksMap(chainId)[address.toLowerCase()] : undefined
 }
+
+export const isHookWhitelisted = memoize(
+  (chainId?: ChainId, address?: Address): boolean => {
+    if (!chainId || !address) return false
+    return (
+      whitelistLabeledHooksList[chainId].some((addr) => isAddressEqual(addr, address)) ||
```

Comment:
this would cause crash if whitelistLabeledHooksList[chainId] returns undefined , should be with optional chaining